code/src/main/kotlin/com/expediagroup/sdk/v2/core/extension/io/InputStreamExtension.kt

```diff
@@ -0,0 +1,21 @@
+package com.expediagroup.sdk.v2.core.extension.io
+
+import okio.Buffer
+import java.io.InputStream
+
+fun InputStream.toBuffer(contentLength: Int, exhaustStream: Boolean = true): Buffer {
+    val buffer = Buffer()
```

Comment:
Can you take a look into `TeeInputStream` (from Apache commons)? this is utility that does what we want to achieve here. 

>The TeeInputStream class from Apache Commons IO duplicates the data read from an input stream to an output stream.
As you read from the InputStream, it writes the data to an OutputStream simultaneously.

It removes the risky logic around mark and reset.

```kotlin
fun getLoggingInputStream(originalInputStream: InputStream): Pair<InputStream, ByteArray> {
    val baos = ByteArrayOutputStream()
    val teeInputStream = TeeInputStream(originalInputStream, baos)
    // Use the teeInputStream for further processing
    return Pair(teeInputStream, baos.toByteArray())
}
```

Reply:
Discussed offline.